guide_drafts/trait_derives.md

```diff
@@ -0,0 +1,954 @@
+Deriving Traits in Depth
+===============
+
+Part of what makes Diesel's query builder so powerful is
+its ability to assist writing safe SQL queries in Rust.
+Diesel enables this level of safety through implementing 
+a series of traits on your structs.
+Writing these implementations by hand can be very laborious,
+so Diesel offers custom derives.
+
+In this guide,
+we will cover each trait in detail in terms of its use cases 
```

Comment:
> each [of these] traits

just a technicality but there are of course more traits that exist in Diesel now or in the future that this guide doesn't cover